### DISCUSSION TITLE
Support for software citation metadata on GitHub?

### DISCUSSION THREAD

**Discussion by sdruskat** [community/community#22415] [New to GitHub]
<p><em>Disclaimer:</em> This adheres mostly to GitHub projects for scientific / research software, but may be of interest to others as well.</p>
<p>In scientific and research communities, the **citation of research software** (i.e., software that’s used in and for research) is an issue that needs to be resolved, as citations are still the common currency for academic credit and acknowledgment.</p>
<p>Last year, the <a href="https://www.force11.org/group/software-citation-working-group" title="FORCE11 Software Citation Working Group homepage" rel="nofollow noopener">FORCE11 Software Citation Working Group</a> has published a <a href="https://peerj.com/articles/cs-86/" title="Software citation principles paper on PeerJ" rel="nofollow noopener">paper on Software Citation Principles</a>, a community effort outlining what the main issues with software citation are, and what best practices can be applied.</p>
<p>The successor of that working group, the <a href="https://www.force11.org/group/software-citation-implementation-working-group" title="FORCE11 Software Citation Implementation Working Group homepage" rel="nofollow noopener">FORCE11 Software Citation Implementation Working Group</a> (also <a href="https://github.com/force11/force11-sciwg" title="https://github.com/force11/force11-sciwg" rel="nofollow noopener">here on GitHub</a>) is currently working on, you will have guessed, implementations for software citation, including the <a href="https://citation-file-format.github.io" title="Citation File Format GitHub pages" rel="nofollow noopener">Citation File Format</a> and <a href="https://codemeta.github.io/" title="CodeMeta GitHub pages" rel="nofollow noopener">CodeMeta</a>.</p>
<p>I’d like to start a conversation of whether and how software citation metadata (provided in either or both of the above formats) could be re-used by the GitHub platform. I’m thinking along the lines of introducing a <em>Cite</em> “tab” - similar to the <a href="https://help.github.com/articles/adding-a-license-to-a-repository/" rel="nofollow noopener"><em>License</em> “tab”</a> on the project site to display a formatted version of the citation metadata by users, e.g., in BibTeX, Endnote, Reference Manager (RIS) formats, and in a plain-text version for copying and pasting (e.g., based on the <a href="https://github.com/citation-style-language" rel="nofollow noopener">Citation Style Language</a>).</p>
<p><em>Disclaimer 2: I’m a member of the FORCE11 SCIWG, the main developer of the Citation File Format, and a contributor of CodeMeta.</em></p>

**Comment by lee-dohm** [comment]
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>

**Comment by arfon** [comment, marked as answer]
<p>Just wanted to add (for anyone stumbling on this historical post) that GitHub does indeed now support software citation files. You can read more about this in the recent blog post: <a href="https://github.blog/2021-08-19-enhanced-support-citations-github/" class="inline-onebox" rel="noopener nofollow ugc">Enhanced support for citations on GitHub | The GitHub Blog</a></p>

**Comment by sdruskat** [comment]
<p>Resolved indeed (high five)!</p>

**Comment by sdruskat** [comment]
<p>Ping: There is a now a website to get you started writing CFF files from scratch: <a href="https://bit.ly/cffinit" class="inline-onebox" rel="noopener nofollow ugc">cffinit</a>.<br>
It’s a web form with validation created by the Netherlands eScience Center.</p>
